src/Simulation/Native/src/simulator/qir.cpp

```diff
@@ -0,0 +1,504 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
(Nit)
If this file is a result of renaming/moving of some other file, then consider renaming/moving with `git mv ..` to preserve the history and minimize the diff.

Reply:
Git rename detection doesn't use `git mv`. It works the same if you `git rm` and `git add` two files with similar contents. The rename is detected at diff time and has a configurable similarity threshold (default 50%); see [`git diff --find-renames[=<n>]`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt--Mltngt). So if more than 50% of the file changed, it won't be seen as a rename by default, even if you used `git mv`.